lib/components/form/mode-buttons.js

```diff
@@ -0,0 +1,75 @@
+import React, { useContext } from 'react'
+
+import Icon from '../narrative/icon'
+import { ComponentContext } from '../../util/contexts'
+
+export const MODE_OPTIONS = [
+  {
+    mode: 'TRANSIT',
+    label: 'Transit'
+  },
+  {
+    mode: 'WALK',
+    label: 'Walking'
+  },
+  {
+    mode: 'CAR',
+    label: 'Drive'
+  },
+  {
+    mode: 'BICYCLE',
+    label: 'Bicycle'
+  },
+  {
+    icon: 'mobile',
+    mode: 'RENT', // TODO: include HAIL?
+    label: 'Rental options'
+  }
+]
+
+const ModeButtons = ({
+  className,
+  endStyle = {},
+  onClick,
+  selectedModes = [],
+  style = {}
+}) => {
+  const {ModeIcon} = useContext(ComponentContext)
+  return MODE_OPTIONS.map((item, index) => {
+    let buttonStyle = style
+    // Add end style to button style if rendering the last item.
+    if (index === MODE_OPTIONS.length - 1) {
+      buttonStyle = {...buttonStyle, ...endStyle}
```

Comment:
Use styled-components for this. (This is the same remark as in the batch routing panel.) In doing so, there is no need to define in the JS code a separate style for the last button.

Reply:
But the layout is very nice.